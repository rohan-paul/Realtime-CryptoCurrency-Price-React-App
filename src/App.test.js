import React from "react"
import renderer from "react-test-renderer"
import { MuiThemeProvider } from "@material-ui/core/styles"
import globalTheme from "./globalTheme"
import PropTypes from "prop-types"
import { mount, shallow } from "enzyme"
import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import { Provider } from "react-redux"
import App from "./App"
import { Action } from "redux"
import { createShallow } from "@material-ui/core/test-utils"
import * as Redux from "react-redux"

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

describe("App Component", () => {
  let shallow: typeof enzyme.shallow
  let useSelectorSpy: jest.SpyInstance
  let useDispatchSpy: jest.SpyInstance

  let store
  let component

  beforeEach(() => {
    shallow = createShallow()
    useSelectorSpy = jest.spyOn(Redux, "useSelector")
    useDispatchSpy = jest.spyOn(Redux, "useDispatch")
    useDispatchSpy.mockImplementation(() => (cb: Action) => cb)

    store = mockStore({
      globalStore: {
        loading: false,
        error_while_fetching_initial_currency_list: false,
        error_while_fetching_initial_data: false,
        city_to_search: "",
        snackbar: false,
        topTenUsersInCity: [],
        totalNoOfUsersFromAPI: 0,
      },
    })
  })

  it("renders without crashing", () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MuiThemeProvider theme={globalTheme}>
          <App />
        </MuiThemeProvider>
      </Provider>,
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("should work", () => {
    useSelectorSpy.mockReturnValue({
      bar: "baz",
    })
    const wrapper = shallow(<App />)
    expect(wrapper.is(App))
  })

  /*   it("renders an error message when a network error occurs", () => {
    props.users.error = true
    const tree = renderer.create(<App {...props} />)

    expect(tree.toJSON()).toMatchSnapshot()
  }) */

  /*   it("calls the getUsers function when the button is clicked", () => {
    props.getUsers = jest.fn()
    const wrapper = shallow(<App {...props} />)
    const spy = jest.spyOn(wrapper.instance().props, "getUsers")

    wrapper.find("button").simulate("click")
    expect(spy).toHaveBeenCalled()
  }) */

  /*   it("renders the User component correctly", () => {
    props.users.users = [
      {
        id: 1,
        name: "foo",
      },
    ]
    const wrapper = shallow(<App {...props} />)

    expect(wrapper.find("User").length).toBe(1)
  }) */
})
