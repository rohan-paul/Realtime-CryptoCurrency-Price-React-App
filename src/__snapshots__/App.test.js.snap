// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div>
  <div
    className="makeStyles-container-1"
  >
    <div
      className="makeStyles-container-4"
    >
      <div
        className="makeStyles-internalContainer-5"
      >
        <div
          className="makeStyles-textContainer-7"
        >
          <h1
            className="MuiTypography-root-8 MuiTypography-h1-13"
          >
             
            Crypto List
             
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-contentContainer-3"
  >
    <div
      className="makeStyles-container-38"
    >
      <div
        className="makeStyles-table-39"
      >
        <div
          className="makeStyles-inputandButtonContainer-40"
        >
          <div
            className="makeStyles-selectionBox-41 css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-yk16xz-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-1hwfws3"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <button
            className="MuiButtonBase-root-73 MuiButton-root-44 MuiButton-contained-52 makeStyles-button-42 MuiButton-containedPrimary-53 MuiButton-containedSizeLarge-64 MuiButton-sizeLarge-66 Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label-45"
            >
              <h3
                className="MuiTypography-root-8 MuiTypography-h3-15"
              >
                Proceed
              </h3>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
